browser/brave_ads/BUILD.gn

```diff
@@ -10,20 +10,42 @@ source_set("brave_ads") {
     "ads_service_factory.h",
     "ads_tab_helper.cc",
     "ads_tab_helper.h",
+    "brave_ads_host_private.cc",
+    "brave_ads_host_private.h",
   ]
 
   deps = [
     "//base",
     "//brave/browser/profiles:util",
     "//brave/components/brave_adaptive_captcha/buildflags",
     "//brave/components/brave_ads/browser",
+    "//brave/components/brave_ads/common:mojom",
+    "//chrome/browser/profiles:profile",
     "//components/dom_distiller/content/browser",
     "//components/keyed_service/content",
     "//components/sessions",
     "//content/public/browser",
     "//ui/base",
   ]
 
+  if (is_android) {
+    sources += [
+      "brave_ads_host_android.cc",
+      "brave_ads_host_android.h",
+    ]
+  } else {
+    sources += [
+      "brave_ads_host.cc",
+      "brave_ads_host.h",
+    ]
+
+    deps += [
+      "//chrome/browser/extensions",
```

Comment:
these are not `is_android` deps, they are guarded by `enable_extensions` so you need `} else {` -> `} else if (enable_extensions) {` around this entire block (including brave_ads_host sources)

Reply:
really this entire target needs to be converted to sources.gni because https://github.com/brave/brave-core/pull/9611/files#diff-4ffcba2467169bfe727203cecc4156ce0aac099aa35abc9e94c497d90c483a40R6 can't be easily resolved

Reply:
this will be fixed in a separate ticket: https://github.com/brave/brave-browser/issues/17663